docarray/array/mixins/io/pushpull.py

```diff
@@ -57,7 +57,9 @@ class PushPullMixin:
 
     _max_bytes = 4 * 1024 * 1024 * 1024
 
-    def push(self, name: str, show_progress: bool = False) -> Dict:
+    def push(
+        self, name: str, *, public: bool = True, show_progress: bool = False
```

Comment:
why did you add * ? are we expecting extra positional args in the future ?

Reply:
Yeah, we might need to add new parameters in the future.

I added `public` before the `show_progress` param because it is more important than `show_progress`.
I think it is more intuitive.

Since `show_progress` is a boolean and was a 2nd positional param, I changed those to keyword params.
Otherwise, the old code will work incorrectly, if `show_progress` was true.

I think it's better to fail than to work incorrectly.
Also, it will be easier to add new parameters in the future. We will just add keyword parameters without worrying about the order.